src/connection.cpp

```diff
@@ -740,14 +865,41 @@ namespace oxen::quic
         return 0;
     }
 
+    // this callback is defined for debugging datagrams
+    int Connection::ack_datagram(uint64_t dgram_id)
+    {
+        log::trace(log_cat, "Connection (CID: {}) acked datagram ID:{}", _source_cid, dgram_id);
+        return 0;
+    }
+
     int Connection::recv_datagram(bstring_view data, bool fin)
     {
         log::trace(log_cat, "Connection (CID: {}) received datagram: {}", _source_cid, buffer_printer{data});
 
-        // append received datagram to tetris_buffer if packet_splitting is enabled
-        if (_packet_splitting)
+        std::optional<bstring> maybe_data;
+        uint16_t dgid = oxenc::load_big_to_host<uint16_t>(data.data());
```

Comment:
When packet splitting is completely disabled on the connection we shouldn't be sending a dgid (to avoid the overhead on datagrams).

Reply:
fixed